src/rust/bin/inspect/transaction.rs

```diff
@@ -0,0 +1,544 @@
+use std::{
+    collections::HashMap,
+    convert::{TryFrom, TryInto},
+};
+
+use bellman::groth16;
+use group::GroupEncoding;
+use orchard::note_encryption::OrchardDomain;
+use secp256k1::{Secp256k1, VerifyOnly};
+use zcash_address::{
+    unified::{self, Encoding},
+    ToAddress, ZcashAddress,
+};
+use zcash_note_encryption::try_output_recovery_with_ovk;
+#[allow(deprecated)]
+use zcash_primitives::{
+    consensus::BlockHeight,
+    legacy::{keys::pubkey_to_address, Script, TransparentAddress},
+    memo::{Memo, MemoBytes},
+    sapling::note_encryption::SaplingDomain,
+    transaction::{
+        components::{orchard as orchard_serialization, sapling, transparent, Amount},
+        sighash::{signature_hash, SignableInput, TransparentAuthorizingContext},
+        txid::TxIdDigester,
+        Authorization, Transaction, TransactionData, TxVersion,
+    },
+};
+use zcash_proofs::sapling::SaplingVerificationContext;
+
+use crate::{context::Context, GROTH16_PARAMS, ORCHARD_VK};
+
+pub fn is_coinbase(tx: &Transaction) -> bool {
+    tx.transparent_bundle()
+        .map(|b| b.is_coinbase())
+        .unwrap_or(false)
+}
+
+pub fn extract_height_from_coinbase(tx: &Transaction) -> Option<BlockHeight> {
+    const OP_0: u8 = 0x00;
+    const OP_1NEGATE: u8 = 0x4f;
+    const OP_1: u8 = 0x51;
+    const OP_16: u8 = 0x60;
+
+    tx.transparent_bundle()
+        .and_then(|bundle| bundle.vin.get(0))
+        .and_then(|input| match input.script_sig.0.get(0).copied() {
+            // {0, -1} will never occur as the first byte of a coinbase scriptSig.
+            Some(OP_0 | OP_1NEGATE) => None,
+            // Blocks 1 to 16.
+            Some(h @ OP_1..=OP_16) => Some(BlockHeight::from_u32((h - OP_1 + 1).into())),
+            // All other heights use CScriptNum encoding, which will never be longer
+            // than 8 bytes for Zcash heights. These have the format
```

Comment:
It's required to be 1..=5 bytes:
![image](https://user-images.githubusercontent.com/643204/178128304-d2c63f20-6120-4104-a32a-23a3e369904d.png)